AlgebraInLean/Chapter02/Sheet04.lean

```diff
@@ -4,196 +4,209 @@ set_option linter.unusedTactic false
 
 namespace AlgebraInLean
 
-variable {G : Type*} [Group G]
-
-/- ## Endomorphisms and Automorphisms -/
-
 /-
-
-In Sheet 1 of this chapter, you were introduced to homomorphisms and isomorphisms. In this
-sheet, we will take a deeper dive with morphisms and some attributes that definitionally
-separate different kids of morphisms.
-
-Particularly, we will begin with endomorphims.
-
-We define an endomorphism to be a homorphism from an object onto itself. In the case of
-`AlgebraInLean`, this means that a _group_ endomporphism is a group homomorphism from an
-arbitrary group G back to itself.
-
-As you have seen previously, in the context of Algebra, "group" is often omitted when discussing
-group endomorphisms. An endomorphism (and morphisms in general) can be defined among many
-different types of mathematical objects, but in AlgebraInLean it will always refer to a group
-endomorphism.
-
-Let's take a look at how this would be defined in Lean:
+Another special structure of morphism is the endomorphism, which is a morphism from an object to
```

Comment:
Lots of rewording here?

Reply:
Yeah. There's no reason for Endomorphism and Automorphism to have their own definitions, since that part is already encoded in the type of the function itself. We also don't really use endomorphisms, so I cut down that part to go straight to automorphisms. I guess we could prove that the set of endomorphisms forms a monoid, but that feels like it's going too far.